src/client/common/amlContext.ts

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { inject, named } from 'inversify';
+import { Memento } from 'vscode';
+import { IExtensionSingleActivationService } from '../activation/types';
+import { setSharedProperty } from '../telemetry';
+import { IWorkspaceService } from './application/types';
+import { GLOBAL_MEMENTO, IMemento } from './types';
+import { noop } from './utils/misc';
+
+const amlComputeMementoKey = 'JVSC_IS_AML_COMPUTE_INSTANCE';
+
+export class AmlComputeContext implements IExtensionSingleActivationService {
+    constructor(
+        @inject(IMemento) @named(GLOBAL_MEMENTO) private readonly memento: Memento,
+        @inject(IWorkspaceService) private readonly workspace: IWorkspaceService
+    ) {}
+
+    public get isAmlCompute() {
+        return this.memento.get<boolean>(amlComputeMementoKey, false) || this.isAmlComputeWorkspace();
+    }
+    private isAmlComputeWorkspace() {
+        let isRemoteConnection: boolean = false;
+        // If there is at least 1 folder, we can check if the scheme and authority match a remote connection
+        if (Array.isArray(this.workspace.workspaceFolders) && this.workspace.workspaceFolders.length > 0) {
+            // We are in a remote connection to a CI if the scheme is vscode-remote and the authority contains amlext
+            isRemoteConnection = this.workspace.workspaceFolders.some(
```

Comment:
Logic courtesy of Shantnu (from AML), @SiddhanthUnnithan

Reply:
Not super happy with this one off. Feels like we should have the AML extension do this but I don't have a good idea for how we might let them make this decision. 

Opening up our experiment API seems kinda like overkill.

Reply:
It does seem a little bit fragile, but aren't we hoping that this all goes away if we can push native notebooks to full usage in a few months?